### PR TITLE
permuteddimsview -> PermutedDimsArray

### PR DESCRIPTION
**Pull request by vtjeng**
## Summary
`permutteddimsview` is deprecated.

https://github.com/JuliaImages/ImageCore.jl/blob/c4d3bc959c1b9fe7fde5986edf443f7773d3943e/src/deprecations.jl#L88

We remove all references to this function (other than in tests specifically for deprecation), replacing it with PermutedDimsarray instead.

As part of this work, we re-ran the code in `docs/src/views.md` and updated it. 

In particular, we remove a section that is now irrelevant ("A note on the return types from the views"). Here is the code that was written for that section:

## Return types from views

```@meta
DocTestSetup = quote
    using Colors, ImageCore
end
```

Consider the following examples with different input images.

In both instances, using the `channelview` returns a `reinterpret`
(but with different type parameters).

```jldoctest
julia> img = [RGB(1,0,0) RGB(0,1,0);
              RGB(0,0,1) RGB(0,0,0)]
2×2 Array{RGB{N0f8},2} with eltype RGB{Normed{UInt8,8}}:
 RGB{N0f8}(1.0,0.0,0.0)  RGB{N0f8}(0.0,1.0,0.0)
 RGB{N0f8}(0.0,0.0,1.0)  RGB{N0f8}(0.0,0.0,0.0)

julia> cv = channelview(img)
3×2×2 reinterpret(N0f8, ::Array{RGB{N0f8},3}):
[:, :, 1] =
 1.0  0.0
 0.0  0.0
 0.0  1.0

[:, :, 2] =
 0.0  0.0
 1.0  0.0
 0.0  0.0
```

```jldoctest
julia> img0 = rand(RGB{Float64}, 3, 2)
3×2 Array{RGB{Float64},2} with eltype RGB{Float64}:
 RGB{Float64}(0.721651,0.436883,0.0249213)  RGB{Float64}(0.838438,0.741055,0.289334)
 RGB{Float64}(0.314968,0.175824,0.392317)   RGB{Float64}(0.984615,0.186958,0.328459)
 RGB{Float64}(0.982128,0.922775,0.704911)   RGB{Float64}(0.967755,0.488064,0.780718)

julia> imgs = view(img0, 1:2:3, :)
2×2 view(::Array{RGB{Float64},2}, 1:2:3, :) with eltype RGB{Float64}:
 RGB{Float64}(0.721651,0.436883,0.0249213)  RGB{Float64}(0.838438,0.741055,0.289334)
 RGB{Float64}(0.982128,0.922775,0.704911)   RGB{Float64}(0.967755,0.488064,0.780718)

julia> channelview(imgs)
3×2×2 reinterpret(Float64, reshape(view(::Array{RGB{Float64},2}, 1:2:3, :), 1, 2, 2)):
[:, :, 1] =
 0.721651   0.982128
 0.436883   0.922775
 0.0249213  0.704911

[:, :, 2] =
 0.838438  0.967755
 0.741055  0.488064
 0.289334  0.780718
 ```

### CODE REVIEW

**Comment by vtjeng** [comment]
@timholy It looks like I'm failing several unit tests here that aren't related to the code change. Specifically, starting at https://github.com/JuliaImages/ImageCore.jl/pull/143/checks?check_run_id=1173227622#step:5:318, 

```
All errors:
===========================================
ArgumentError: Package ImageIO not found in current path:
- Run `import Pkg; Pkg.add("ImageIO")` to install the ImageIO package.

===========================================
ArgumentError: Package QuartzImageIO not found in current path:
- Run `import Pkg; Pkg.add("QuartzImageIO")` to install the QuartzImageIO package.

===========================================

UndefVarError: libwand not defined
===========================================
Fatal error:
4D inputs: Error During Test at /Users/runner/work/ImageCore.jl/ImageCore.jl/test/views.jl:235
  Got exception outside of a @test
  ArgumentError: Package ImageIO not found in current path:
  - Run `import Pkg; Pkg.add("ImageIO")` to install the ImageIO package.
...
```

Do you know what's going on here?

**Comment by johnnychen94** [comment]
This is due to the recent changes in BinaryBuilder and artifacts in julialang/julia, see also https://github.com/JuliaIO/ImageMagick.jl/issues/189

I've disabled fail-fast in 16c688c08c4c5734c72184af6cc27937564a91f6, now I need to close and reopen to retrigger the test on the new base.

File: docs/src/views.md [hunk 1]
```diff
@@ -244,86 +247,3 @@ large images.
 
 `colorview`'s ability to combine multiple grayscale images is based on
 another view, `StackedView`, which you can also use directly.
-
-## A note on the return types from the views
```
Comment (johnnychen94) [inline docs/src/views.md:248]: I think this section is still useful, might just need to use `channelview` instead.

see also: https://github.com/JuliaImages/juliaimages.github.io/issues/154

**Review by johnnychen94 — COMMENTED** [review]
Thanks for updating this.

I guess this documentation was written when `@example` feature is not yet implemented, would you mind to upgrade the code blocks with [`@example`](https://juliadocs.github.io/Documenter.jl/stable/man/syntax/#@example-block) and [`jldoctest`](https://juliadocs.github.io/Documenter.jl/stable/man/doctests/) things? That could indeed help capture such kind of outdated docs in the future.

File: docs/src/views.md (hunk 1 above)
Comment (vtjeng) [reply docs/src/views.md:248]: @johnnychen94 it looks like the examples below were explaining how the lowercase `channelview` returned different types sometimes (which is no longer true, at least for the examples that are provided here). Is that you would like to document here that the input images can have different types but that `channelview` standardizes it?

Comment (johnnychen94) [reply docs/src/views.md:248]: Thanks for explaining; I failed to get the context of this section. Yes, it makes sense to remove it.

File: docs/src/reference.md [hunk 2]
```diff
@@ -14,7 +14,6 @@ channelview
 colorview
 rawview
 normedview
-permuteddimsview
```
Comment (vtjeng) [inline docs/src/reference.md:17]: Removing as `PermutedDimsArray` is not defined in this repo.

**Comment by vtjeng** [comment]
Okay, I've updated this and tested the doctests locally with

```
cd docs
julia --project --color=yes make.jl
```

This should be ready for another review.

**Comment by johnnychen94** [comment]
@vtjeng thank you very much for these updates!

**Pull request merged**